### DISCUSSION TITLE
Internationalization MIDDLEWARE Problems

### DISCUSSION THREAD

**Discussion by rossark0** [fuma-nama/fumadocs#389] [Q&A]
So, I'm trying to integrate fumadocs with an existing project of mine. I already had 2 middlewares working together (CLERK auth and NEXT-INTL) how can I implement the i18n to my fumadocs docs with those others middlewares? Or do I need to remove NEXT-INTL from my project?

=================================================================================================

I am getting this error 

<img width="1185" alt="Screenshot 2024-05-21 at 13 11 24" src="https://github.com/fuma-nama/fumadocs/assets/35125885/c817bc9b-3f43-425f-ad1c-612e5d18dae7"> 

once I put this on my middleware.ts file: 

const translatedMiddleDocs = createI18nMiddleware({
  languages: ['en', 'pt'],

  // Used when no locale matches
  defaultLanguage: 'pt',
})

===========================================================================================

Also in my docs layout file I am getting this:

<img width="811" alt="Screenshot 2024-05-21 at 13 17 37" src="https://github.com/fuma-nama/fumadocs/assets/35125885/535236de-b8b6-4675-a0b5-7e7f7093d262">

===========================================================================================

Here's my source.ts file: 

<img width="853" alt="Screenshot 2024-05-21 at 13 20 21" src="https://github.com/fuma-nama/fumadocs/assets/35125885/4fd73e57-9530-4057-854f-93c291c8bd3c">

**Comment by fuma-nama** [comment]
1. Can you try naming it `middleware.mts`?
2. Following the docs, you have to use `pageTree[locale]` to access the page tree of a language.

**Comment by fuma-nama** [comment]
Fixed some ESM compatibility problems with Next.js, upgrading fumadocs may fix your problems
